packages/peregrine/lib/hooks/useWindowSize.js

```diff
@@ -22,9 +23,9 @@ const getSize = () => {
 const useWindowSizeListener = () => {
     const [windowSize, setWindowSize] = useState(getSize());
 
-    const handleResize = () => {
+    const handleResize = useCallback(() => {
         setWindowSize(getSize());
-    };
+    }, [setWindowSize]);
 
     useEventListener(window, 'resize', handleResize);
```

Comment:
Is this change needed?
```suggestion
    useEventListener(globalThis, 'resize', handleResize);
```